plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+"""
+The nxos bgp_address_family fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+
+from copy import deepcopy
+
+from ansible.module_utils.six import iteritems
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import (
+    utils,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.rm_templates.bgp_address_family import (
+    Bgp_address_familyTemplate,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.argspec.bgp_address_family.bgp_address_family import (
+    Bgp_address_familyArgs,
+)
+
+
+class Bgp_address_familyFacts(object):
+    """ The nxos bgp_address_family facts class
+    """
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Bgp_address_familyArgs.argument_spec
+
+    def get_config(self, connection):
+        """Wrapper method for `connection.get()`
+        This method exists solely to allow the unit test framework to mock device connection calls.
+        """
+        return connection.get("show running-config | section '^router bgp'")
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for Bgp_address_family network resource
+
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+
+        :rtype: dictionary
+        :returns: facts
+        """
+        facts = {}
+        objs = []
+
+        if not data:
+            data = self.get_config(connection)
+
+        data = self._flatten_config(data)
+
+        # parse native config using the Bgp_address_family template
+        bgp_address_family_parser = Bgp_address_familyTemplate(
+            lines=data.splitlines()
+        )
+        objs = bgp_address_family_parser.parse()
+        if objs:
+            nbr = []
+            if "address_family" in objs:
+                # remove neighbor AF entries
+                for k, v in iteritems(objs["address_family"]):
+                    if not k.startswith("nbr_"):
+                        nbr.append(k)
+                for x in nbr:
+                    del objs["address_family"][x]
+                objs["address_family"] = list(objs["address_family"].values())
+                # TO-DO sort all list of dictionaries
```

Comment:
looks like some expected modification.

Reply:
Yeah, I kept a reminder to sort every list of dictionary. Will push a commit.